src/modules/launcher/Plugins/Microsoft.Plugin.Calculator/CalculateHelper.cs

```diff
@@ -11,10 +11,10 @@ public static class CalculateHelper
     {
         private static readonly Regex RegValidExpressChar = new Regex(
             @"^(" +
-            @"ceil|floor|exp|pi|e|max|min|det|abs|log|ln|sqrt|" +
-            @"sin|cos|tan|arcsin|arccos|arctan|" +
-            @"eigval|eigvec|eig|sum|polar|plot|round|sort|real|zeta|" +
-            @"bin2dec|hex2dec|oct2dec|" +
+            @"ceil\s*\(|floor\s*\(|exp\s*\(|max\s*\(|min\s*\(|abs\s*\(|log\s*\(|ln\s*\(|sqrt\s*\(|pow\s*\(|" +
```

Comment:
Is there any reason for removing `e` and `det` from regex?

Reply:
`det` is not a valid input to the mages library. An input with `det` to mages would throw an exception. 
This is a list of all the valid functions that mages supports - https://github.com/FlorianRappl/Mages/blob/master/doc/functions.md

Regarding `e` - It was duplicated in the regex before the changes in this PR. One of them has been removed in this PR.
Note the e in line 19 that is still present. Added a unit test as well to validate that it's still a part of the regex.

![image](https://user-images.githubusercontent.com/28739210/96757654-95870300-138a-11eb-8c5a-5320d63b5f01.png)

Reply:
`exp` and `rand` are also duplicated on line 15.